src/shared/utils/validateExplicitKey.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule validateExplicitKey
+ */
+
+/**
+ * ReactElementValidator provides a wrapper around a element factory
+ * which validates the props passed to the element. This is intended to be
+ * used only in DEV and could be replaced by a static type checker for languages
+ * that support it.
+ */
+
+'use strict';
+
+var ReactCurrentOwner = require('ReactCurrentOwner');
+var getComponentName = require('getComponentName');
+var getDeclarationErrorAddendum = require('getDeclarationErrorAddendum');
+
+if (__DEV__) {
+  var warning = require('fbjs/lib/warning');
+  var {
+    getCurrentStackAddendum,
+  } = require('ReactComponentTreeHook');
+}
+
+/**
+ * Warn if there's no key explicitly set on dynamic arrays of children or
+ * object keys are not valid. This allows us to keep track of children between
+ * updates.
+ */
+var ownerHasKeyUseWarning = {};
+
+function getCurrentComponentErrorInfo(parentType) {
+  if (!parentType) {
+    return '';
+  }
```

Comment:
Making this return an empty string when no parentType is passed is kind of a hack. Ideally would refactor this, as I explained in the commit message for this change.

Also planning to open a 'good-first-bug' issue to get this flow typed.